bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/AbstractWeatherHandler.java

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.fmiweather.internal;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.measure.Quantity;
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.fmiweather.internal.client.Client;
+import org.openhab.binding.fmiweather.internal.client.Data;
+import org.openhab.binding.fmiweather.internal.client.FMIResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AbstractWeatherHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Sami Salonen - Initial contribution
+ */
+@NonNullByDefault
+public abstract class AbstractWeatherHandler extends BaseThingHandler {
+
+    private static final ZoneId UTC = ZoneId.of("UTC");
+    protected static final String PROP_LONGITUDE = "longitude";
+    protected static final String PROP_LATITUDE = "latitude";
+    protected static final String PROP_NAME = "name";
+    protected static final String PROP_REGION = "region";
+    private static final long REFRESH_THROTTLE_MILLIS = 10_000;
+
+    protected static final int TIMEOUT_MILLIS = 30_000;
```

Comment:
Can you make this way shorter to not block the scheduler thread for so long time, when the HTTP request times out?

Reply:
Reduced to 10 s: 44407132cec7a941561e50135cb85930afa90440